ledger/error/src/main/scala/com/daml/error/generator/ErrorCodeDocumentationGenerator.scala

```diff
@@ -104,38 +106,52 @@ class ErrorCodeDocumentationGenerator(prefixes: Array[String] = Array("com.daml"
             Some(value)
         }
 
+      val existingDeprecation = state.deprecation
+      val updatedDeprecationString =
+        updateString(existingDeprecation.map(_.deprecation), updatedDeprecation, "deprecation")
       val existingExplanation = state.explanation
       val updatedExplanationString =
         updateString(existingExplanation.map(_.explanation), updatedExplanation, "explanation")
       val existingResolution = state.resolution
       val updatedResolutionString =
         updateString(existingResolution.map(_.resolution), updatedResolution, "resolution")
       GetAnnotationsState(
+        updatedDeprecationString.map(Deprecation),
         updatedExplanationString.map(Explanation),
         updatedResolutionString.map(Resolution),
       )
     }
 
-    val doc = annotations.foldLeft(GetAnnotationsState(None, None)) { case (state, annotation) =>
-      if (isAnnotation(annotation, deprecatedTypeName))
-        state
-      else if (isAnnotation(annotation, explanationTypeName))
-        update(state, updatedExplanation = Some(parseAnnotationValue(annotation.tree)))
-      else if (isAnnotation(annotation, resolutionTypeName))
-        update(state, updatedResolution = Some(parseAnnotationValue(annotation.tree)))
-      else
-        sys.error(
-          s"Unexpected annotation detected (${annotations.map(annotationTypeName)} but the only supported ones are $acceptedTypeNames)."
-        )
+    val doc = annotations.foldLeft(GetAnnotationsState(None, None, None)) {
+      case (state, annotation) =>
+        if (isAnnotation(annotation, deprecatedTypeName))
+          update(state, updatedDeprecation = Some(parseDeprecatedAnnotationValue(annotation.tree)))
+        else if (isAnnotation(annotation, explanationTypeName))
+          update(state, updatedExplanation = Some(parseAnnotationValue(annotation.tree)))
+        else if (isAnnotation(annotation, resolutionTypeName))
+          update(state, updatedResolution = Some(parseAnnotationValue(annotation.tree)))
+        else
+          sys.error(
+            s"Unexpected annotation detected (${annotations.map(annotationTypeName)} but the only supported ones are $acceptedTypeNames)."
+          )
     }
 
-    ErrorDocumentationAnnotations(doc.explanation, doc.resolution)
+    ErrorDocumentationAnnotations(doc.deprecation, doc.explanation, doc.resolution)
   }
 
+  private def parseDeprecatedAnnotationValue(tree: ru.Tree): String =
```

Comment:
wow

Reply:
Yeah, Java reflection would be easier but it doesn't seem to help with Scala's `@deprecated` (the properties are not accessible).